ouroboros-consensus/docs/VersioningSchemeDecision.md

```diff
@@ -0,0 +1,286 @@
+# Choosing a Versioning Schemes
+
+This document records our discussions around choosing how to version our packages.
+
+In its current state, the document is primarily focused on one question: for each individual package, how should the version listed in the `.cabal` file on the main integration branch relate to the version in the `.cabal` file in the latest release?
+In particular, I think this question is almost entirely independent of usings `git` branches versus `git` tags for releases, SemVer versus PVP, etc.
+This document should eventually specify that level of detail as well.
+
+Given its current focus, without loss of generality, suppose the repository contains only one package while considering the proposals below.
+
+## Desiderata
```

Comment:
Should scalability be one of the desiderata? Does a certain proposal scale to 20 commits per day, and multiple releases per week?

If a proposal requires each PR to set the `main` version based on either the current `main` version of HEAD, or the latest release version, it probably doesn't.

Reply:
That's a great point, yes!

I just added commit 217f273b2c9d06c5c4dc59690ad3b5d9c07d952f, which favors that desideratum, I think.